openmls/src/group/errors.rs

```diff
@@ -8,269 +8,232 @@ use crate::{
+    InvalidPublicGroupStateSignature,
+    #[error("A duplicate ratchet tree was found.")]
+    DuplicateRatchetTreeExtension,
+    #[error("Error creating external commit.")]
+    CommitError,
+    /// This error indicates the public tree is invalid. See [`PublicTreeError`] for more details.
+    #[error(transparent)]
+    PublicTreeError(#[from] PublicTreeError),
 }
 
-implement_error! {
-    pub enum StageCommitError {
-        Simple {
-            EpochMismatch =
-                "Couldn't stage Commit. The epoch of the group context and MlsPlaintext didn't match.",
-            WrongPlaintextContentType =
-                "stage_commit was called with an MlsPlaintext that is not a Commit.",
-            SelfRemoved =
-                "Tried to stage a commit to a group we are not a part of.",
-            PathKeyPackageVerificationFailure =
-                "Unable to verify the key package signature.",
-            NoParentHashExtension =
-                "Parent hash extension is missing.",
-            ParentHashMismatch =
-                "Parent hash values don't match.",
-            RequiredPathNotFound =
-                "Unable to determine commit path.",
-            ConfirmationTagMissing =
-                "Confirmation Tag is missing.",
-            ConfirmationTagMismatch =
-                "Confirmation tag is invalid.",
-            MissingOwnKeyPackage =
-                "No key package provided to apply own commit.",
-            MissingProposal =
-                "The proposal queue is missing a proposal for the commit.",
-            OwnKeyNotFound =
-                "Missing own key to apply proposal.",
-            InconsistentSenderIndex =
-                "External Committer used the wrong index.",
-            LibraryError =
-                "An unrecoverable error has occurred due to a bug in the implementation.",
-        }
-        Complex {
-            PlaintextSignatureFailure(VerificationError) =
-                "MlsPlaintext signature is invalid.",
-            CodecError(TlsCodecError) =
-                "Tls (de)serialization error occurred.",
-            KeyScheduleError(KeyScheduleError) =
-                "An error occurred in the key schedule.",
-        }
-    }
+/// Stage Commit error
+#[derive(Error, Debug, PartialEq, Clone)]
+pub enum StageCommitError {
+    #[error(transparent)]
+    LibraryError(#[from] LibraryError),
+    #[error("The epoch of the group context and MlsPlaintext didn't match.")]
+    EpochMismatch,
+    #[error("stage_commit was called with an MlsPlaintext that is not a Commit.")]
+    WrongPlaintextContentType,
+    #[error("Unable to verify the key package signature.")]
+    PathKeyPackageVerificationFailure,
+    #[error("Unable to determine commit path.")]
+    RequiredPathNotFound,
+    #[error("Confirmation Tag is missing.")]
+    ConfirmationTagMissing,
+    #[error("Confirmation tag is invalid.")]
```

Comment:
Some messages are real sentences already. I think we should make all of them proper english.
```suggestion
    #[error("The confirmation tag is invalid.")]
```